contracts/src/tokens/memecoin.cairo

```diff
@@ -159,8 +163,11 @@ mod UnruggableMemecoin {
         }
 
         fn transfer(ref self: ContractState, recipient: ContractAddress, amount: u256) -> bool {
-            self._check_max_buy_percentage(amount);
             let sender = get_caller_address();
+            let locker = self.locker_contract.read();
+            if (sender != locker && recipient != locker) {
```

Comment:
Can you move this check inside the _check_max_buy_percentage function ?

Reply:
Sure